pkg/controller/mysqlcluster/internal/syncer/statefullset.go

```diff
@@ -449,6 +451,17 @@ func (s *sfsSyncer) ensureContainersSpec() []core.Container {
 
 func (s *sfsSyncer) ensureVolumes() []core.Volume {
 	fileMode := int32(0644)
+	dataVolume := core.VolumeSource{
+		EmptyDir: s.cluster.Spec.VolumeSpec.EmptyDir,
+		HostPath: s.cluster.Spec.VolumeSpec.HostPath,
+	}
+
+	if s.cluster.Spec.VolumeSpec.PersistentVolumeClaim != nil {
```

Comment:
Allow only one volume in `dataVolume`. For example, if user specifies both, `emptyDir` amd `hostPath`, the `dataVolume` should contain only `hostPath`.